test/test_shuffle_128.cpp

```diff
@@ -0,0 +1,95 @@
+/***************************************************************************
+ *                                                                          *
+ * Copyright (c) QuantStack                                                 *
+ *                                                                          *
+ * Distributed under the terms of the BSD 3-Clause License.                 *
+ *                                                                          *
+ * The full license is in the file LICENSE, distributed with this software. *
+ ****************************************************************************/
+#include "test_utils.hpp"
+
+namespace xsimd
+{
+    template <typename T, std::size_t N>
+    struct init_shuffle_128_base
+    {
+        using shuffle_vector_type = std::array<T, N>;
+        shuffle_vector_type lhs_in, rhs_in, exp_lo, exp_hi;
+
+        std::vector<shuffle_vector_type> create_vectors()
+        {
+            std::vector<shuffle_vector_type> vects;
+            vects.reserve(4);
+
+            /* Generate input data: lhs, rhs */
+            for (size_t i = 0; i < N; ++i)
+            {
+                lhs_in[i] = 2*i + 1;
+                rhs_in[i] = 2*i + 2;
+            }
+            vects.push_back(std::move(lhs_in));
+            vects.push_back(std::move(rhs_in));
+
+            /* Expected shuffle data */
+            for (size_t i = 0, j= 0; i < N/2; ++i, j=j+2)
+            {
+                exp_lo[j] = lhs_in[i];
+                exp_hi[j] = lhs_in[i + N/2];
+
+                exp_lo[j + 1] = rhs_in[i];
+                exp_hi[j + 1] = rhs_in[i + N/2];
+            }
+            vects.push_back(std::move(exp_lo));
+            vects.push_back(std::move(exp_hi));
+
+            return vects;
+        }
+    };
+}
+
+template <class B>
+class shuffle_128_test : public testing::Test
+{
+  protected:
+    using batch_type = B;
+    using value_type = typename B::value_type;
+    static constexpr size_t size = B::size;
+
+    shuffle_128_test()
+    {
+        std::cout << "shuffle-128 test" << std::endl;
+    }
+
+    void shuffle_128_low_high()
+    {
+        auto shuffle_base = xsimd::init_shuffle_128_base<value_type, size>{};
```

Comment:
`xsimd::init_shuffle_128_base<value_type, size> shuffle_base;` should fix the compilation issue.